### PR TITLE
[Component] Additional includes support azure devops artifacts

### PR DESCRIPTION
**Pull request by lalala123123**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_additional_includes.py [hunk 1]
```diff
@@ -201,3 +210,87 @@ def cleanup(self) -> None:
...
+            additional_includes:
+             - your/local/path
+             - type: artifact
+               organization: devops_organization
+               project: devops_project
+               feed: artifacts_feed_name
+               name: universal_package_name
+               version: package_version
+               scope: scope_type
+        If will get the artifacts package from devops to the local, and merge them with the local path into
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_internal/entities/_additional_includes.py:238]: If or It?

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py [hunk 2]
```diff
@@ -470,6 +470,15 @@ def _to_rest_object(self) -> JobBase:
         # This will be resolved in job_operations _resolve_arm_id_or_upload_dependencies.
         component_id = self.component if isinstance(self.component, str) else self.component.id
 
+        # MFE not support pass None or empty input value. Remove the empty inputs and parameters in jobs.
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py:473]: created a task to server to accept binding to empty inputs

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py [hunk 3]
```diff
@@ -470,6 +470,15 @@ def _to_rest_object(self) -> JobBase:
...
+        empty_inputs = [k for k, v in built_inputs.items() if v is None or v == ""]
+        if empty_inputs:
+            built_inputs = {k: v for k, v in built_inputs.items() if k not in empty_inputs}
+            empty_inputs_value = [f"${{{{parent.inputs.{k}}}}}" for k in empty_inputs]
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py:477]: add test case for empty pipeline job input

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py (hunk 2 above)
Comment (lalala123123) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py]: Created a task to backend

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py (hunk 3 above)
Comment (lalala123123) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/pipeline_job.py]: Add test case for empty job input

**Review by wangchao1230 — APPROVED** [review]
